### PR TITLE
Build documentation more strictly

### PR DESCRIPTION
**Pull request by devmotion**
This PR ensures that failures when building the documentation and in doctests actually cause a failing build. It contains some of the fixes in https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/232 and takes inspiration from AbstractGPs.

### CODE REVIEW

**Comment by willtebbutt** [comment]
This looks good. I wonder whether we shouldn't also run the doctests as part of the usual testing cycle -- i.e. when `test/runtests.jl` is run? This would give you more feedback during the development cycle.

**Comment by devmotion** [comment]
I thought about it (e.g., they are included as a separate test set in AbstractGPs) but assumed that it would be sufficient if the build errors if they fail (which is not the case in AbstractGPs currently). I don't have a strong opinion though, I'm fine with adding it to the test pipeline as well.

**Comment by willtebbutt** [comment]
I agree that it is sufficient for the sake of getting CI to complain when the doctests fail, I'm just thinking about the overall development cycle. If you're happy to add it to the test pipeline, that would be great.

**Comment by devmotion** [comment]
OK, done.

File: test/Project.toml [hunk 1]
```diff
@@ -17,6 +18,7 @@ Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f"
 [compat]
 AxisArrays = "0.4.3"
 Distances = "0.9, 0.10"
+Documenter = "0.25"
```
Comment (devmotion) [inline test/Project.toml:21]: JuliaFormatter is not compatible with Documenter 0.26

Comment (willtebbutt) [reply test/Project.toml:21]: Booooo

**Review by willtebbutt — APPROVED** [review]
I'm happy with this now. I'll leave it up to you to figure out what to do about the compat issues.

**Comment by devmotion** [comment]
I opened a PR (https://github.com/domluna/JuliaFormatter.jl/pull/354) but for the tests it should not matter. The webpage is still built with the most recent version since it uses a different project environment.

**Pull request merged**